secret_santa/api/models.py

```diff
@@ -1,3 +1,84 @@
+import random
+import uuid
+from enum import Enum
+
 from django.db import models
 
 # Create your models here.
+
+
+class StatusChoices(Enum):
+    OPEN = "OPEN"
+    MOVING = "MOVING"
+    CLOSE = "CLOSE"
+
+    @classmethod
+    def choices(cls):
+        return tuple((i.name.lower(), i.value) for i in cls)
+
+
+class Users(models.Model):
+    id = models.UUIDField(default=uuid.uuid4, primary_key=True, editable=False, unique=True)
+    first_name = models.CharField(max_length=50)
+    last_name = models.CharField(max_length=50)
+    email = models.CharField(max_length=100)
+
+    def __str__(self):
+        return f"{self.first_name} {self.last_name}"
+
+
+class Boxes(models.Model):
+    id = models.UUIDField(default=uuid.uuid4, primary_key=True, editable=False, unique=True)
+    name_box = models.CharField(max_length=100)
+    create_date = models.DateTimeField(auto_now_add=True)
+    end_date = models.DateTimeField(blank=True, null=True)
+    status_box = models.CharField(max_length=10, choices=StatusChoices.choices(), default="open")
+
+    def __str__(self):
+        return self.name_box
+
+    @property
+    def users(self):
+        giftrequests = self.giftrequests.all()
+
+        users = []
+        for giftrequest in giftrequests:
+            users.append(giftrequest.user)
+        return users
+
+    def move_bag(self):
+        if self.status_box == "moving":
+            giftrequests = self.giftrequests.all()
+            users = self.users
+            for giftrequest in giftrequests:
+                secret_user = giftrequest.user
+                while giftrequest.user == secret_user:
+                    secret_user = random.choice(users)
```

Comment:
If we only have 1 user, will we repeat this loop infinite number of times?

Reply:
this function will start only  if self.status_box == "open" and len(self.users) > 2
i described it in the function below